src/components/codeBlock.tsx

```diff
@@ -106,52 +106,154 @@ type KeywordSelectorProps = {
 };
 
 function OrgAuthTokenCreator() {
-  const codeContext = useContext(CodeContext);
+  const [tokenState, setTokenState] = useState<
+    | {status: 'none'}
+    | {status: 'loading'}
+    | {status: 'success'; token: string}
+    | {status: 'error'}
+  >({status: 'none'});
+  const [isOpen, setIsOpen] = useState(false);
+  const [referenceEl, setReferenceEl] = useState<HTMLSpanElement>(null);
+  const [dropdownEl, setDropdownEl] = useState<HTMLElement>(null);
+  const {codeKeywords} = useContext(CodeContext);
+  const {styles, state, attributes} = usePopper(referenceEl, dropdownEl, {
+    placement: 'bottom',
+    modifiers: [
+      {
+        name: 'offset',
+        options: {offset: [0, 10]},
+      },
+      {name: 'arrow'},
+    ],
+  });
 
-  const [tokenState, setTokenState] = useState<'none' | 'loading' | 'success' | 'error'>(
-    'none'
-  );
-  const [token, setToken] = useState(null);
-  const [sharedSelection] = codeContext.sharedKeywordSelection;
-  const {codeKeywords} = codeContext;
+  useOnClickOutside({
+    ref: {current: referenceEl},
+    enabled: isOpen,
+    handler: () => setIsOpen(false),
+  });
 
-  const choices = codeKeywords?.PROJECT;
+  const createToken = async (orgSlug: string) => {
+    setTokenState({status: 'loading'});
+    const token = await createOrgAuthToken({
+      orgSlug,
+      name: `Generated by Docs on ${new Date().toISOString().slice(0, 10)}`,
+    });
+
+    if (token) {
+      setTokenState({
+        status: 'success',
+        token,
+      });
+    } else {
+      setTokenState({
+        status: 'error',
+      });
+    }
+  };
+
+  const orgSet = new Set<string>();
+  codeKeywords?.PROJECT?.forEach(projectKeyword => {
+    orgSet.add(projectKeyword.ORG_SLUG);
+  });
+  const orgs = [...orgSet];
```

Comment:
So it is clear further down the component that those are the slugs and not whole objects.
```suggestion
  const orgSlugs = [...orgSet];
```